### PR TITLE
feat: add vite specific tags

### PR DESCRIPTION
**Pull request by Julien-R44**
- Add `@vite` tag that which will inject the @vite/client script needed by the HMR
- Add `@viteReactRefresh` tag that will inject the script needed for HMR to work with React, as asked to do here : https://vitejs.dev/guide/backend-integration.html

The tests will fail because it will be necessary to merge and update @adonisjs/application and @adonisjs/core

Related to https://github.com/adonisjs/core/issues/3953

### CODE REVIEW

File: src/Bindings/Vite.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+/*
+ * @adonisjs/view
+ *
+ * (c) Harminder Virk <virk@adonisjs.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+import { ViewContract } from '@ioc:Adonis/Core/View'
```
Comment (RomainLanz) [inline src/Bindings/Vite.ts]:
```suggestion
import type { ViewContract } from '@ioc:Adonis/Core/View'
```

File: providers/ViewProvider.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ import type { RouterContract } from '@ioc:Adonis/Core/Route'
 import type { ApplicationContract } from '@ioc:Adonis/Core/Application'
 import type { AssetsManagerContract } from '@ioc:Adonis/Core/AssetsManager'
 import type { HttpContextConstructorContract } from '@ioc:Adonis/Core/HttpContext'
+import { defineViteDriverBindings } from '../src/Bindings/Vite'
```
Comment (RomainLanz) [inline providers/ViewProvider.ts]: This import should be before types imports

**Review by RomainLanz — CHANGES_REQUESTED** [review]

**Comment by Julien-R44** [comment]
Still needs to merge and update "application" and "core" to see green tests

**Comment by Julien-R44** [comment]
Ready to merge

**Pull request merged**